### ISSUE TITLE
No out put generated 

### ISSUE THREAD

**Issue by bony45** [Hoohm/CITE-seq-Count#135]
Dear all, 
I am not able to generate any out put after running script for some time. 

I am running following command 
CITE-seq-Count -R1 Ext-20-003-001_S1_L006_R1_001.fastq.gz -R2 Ext-20-003-001_S1_L006_R2_001.fastq.gz -t tag.txt -cbf 1 -cbl 16 -umif 17 -umil 26 -cells 10000 -o ext_001_demultiplex
 I donot see any output fine even after script ran for 24 hours. 
Looking forward for suggestions and remedies. 
Thanks 

Mapping done for process 21386. Processed 24,252,023 reads
Processed 1,000,000 reads in 26.11 seconds. Total reads: 18,000,000 in child 21387
Processed 1,000,000 reads in 23.64 seconds. Total reads: 10,000,000 in child 21388
Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.6/site-packages/multiprocess/pool.py", line 479, in _handle_results
    cache[job]._set(i, obj)
  File "/usr/local/lib/python3.6/site-packages/multiprocess/pool.py", line 651, in _set
    self._error_callback(self._value)
TypeError: '_io.TextIOWrapper' object is not callable

Processed 1,000,000 reads in 26.37 seconds. Total reads: 3,000,000 in child 21389
Processed 1,000,000 reads in 22.52 seconds. Total reads: 11,000,000 in child 21388
Processed 1,000,000 reads in 24.88 seconds. Total reads: 19,000,000 in child 21387
Processed 1,000,000 reads in 16.62 seconds. Total reads: 4,000,000 in child 21389
Processed 1,000,000 reads in 22.21 seconds. Total reads: 12,000,000 in child 21388
Processed 1,000,000 reads in 22.65 seconds. Total reads: 20,000,000 in child 21387
Processed 1,000,000 reads in 23.98 seconds. Total reads: 5,000,000 in child 21389
Processed 1,000,000 reads in 21.34 seconds. Total reads: 13,000,000 in child 21388
Processed 1,000,000 reads in 22.82 seconds. Total reads: 21,000,000 in child 21387
Processed 1,000,000 reads in 19.9 seconds. Total reads: 6,000,000 in child 21389
Processed 1,000,000 reads in 17.91 seconds. Total reads: 14,000,000 in child 21388
Processed 1,000,000 reads in 21.08 seconds. Total reads: 22,000,000 in child 21387
Processed 1,000,000 reads in 22.17 seconds. Total reads: 7,000,000 in child 21389
Processed 1,000,000 reads in 21.18 seconds. Total reads: 15,000,000 in child 21388
Processed 1,000,000 reads in 20.01 seconds. Total reads: 23,000,000 in child 21387
Processed 1,000,000 reads in 24.02 seconds. Total reads: 8,000,000 in child 21389
Processed 1,000,000 reads in 23.48 seconds. Total reads: 16,000,000 in child 21388
Processed 1,000,000 reads in 23.65 seconds. Total reads: 24,000,000 in child 21387
Mapping done for process 21387. Processed 24,252,023 reads
 After addition for few more processes  script do not do anything for long time and stay inactive with low or no load. 
Thanks

**Comment by Hoohm** [comment]
Could you maybe provide more cores to run it on? Maybe 12?

**Comment by bony45** [comment]
could you tell me how to set a specific number of core when running this script. I tried -p option but came up with error . 
CITE-seq-Count: error: unrecognized arguments: -p 12

i tried 
CITE-seq-Count -p 12  -R1 Ext-20-003-001_S1_L006_R1_001.fastq.gz -R2 Ext-20-003-001_S1_L006_R2_001.fastq.gz -t tag.txt -cbf 1 -cbl 16 -umif 17 -umil 26 -cells 10000 -o ext_001_demultiplex

**Comment by bony45** [comment]
i tried 16 core . same issue is persisting 
nothing happening after message 
Mapping done for process 24435. Processed 24,252,036 reads

and now Cite-Seq script is using only 0.3 CPU
I do not see any out-put  file in my working directory.

**Comment by Hoohm** [comment]
The option for cores is :

   - [Optional] How many threads/cores should be used by CITE-seq-Count.

-T N_THREADS, --threads N_THREADS, default Number of available cores


What kind of computer are you using?

**Comment by bony45** [comment]
I am using a Linux server with 256 GB ram and 16 cores.
Thanks

Bony De Kumar PhD
Director, COBRE Genomics Core
Rm W127
University of North Dakota
School of Medicine and Health Sciences
Cell Phone: 816-462-8083
Lab Phone: 701-777-6612
Email: bony.dekumar@und.edu

**Comment by Hoohm** [comment]
From the last log you sent, the number of cores didn't change as it was
still ~26000000 reads per process

**Comment by bony45** [comment]
Run start with 16 cores . as you can see , I just started this run.
Thanks

Bony De Kumar PhD
Director, COBRE Genomics Core
Rm W127
University of North Dakota
School of Medicine and Health Sciences
Cell Phone: 816-462-8083
Lab Phone: 701-777-6612
Email: bony.dekumar@und.edu

**Comment by bony45** [comment]
Bony De Kumar PhD
Director, COBRE Genomics Core
Rm W127
Screenshot showing as run is proceeding further.
Thanks

University of North Dakota
School of Medicine and Health Sciences
Cell Phone: 816-462-8083
Lab Phone: 701-777-6612
Email: bony.dekumar@und.edu

**Comment by bony45** [comment]
I tried rerunning in our HPC server. Now it is completing running but before writing output showing following error 
Mapping done for process 3076101. Processed 10,778,677 reads
Processed 1,000,000 reads in 8.116 seconds. Total reads: 8,000,000 in child 3076102
Processed 1,000,000 reads in 9.126 seconds. Total reads: 9,000,000 in child 3076102
Processed 1,000,000 reads in 8.826 seconds. Total reads: 10,000,000 in child 3076102
Mapping done for process 3076102. Processed 10,778,686 reads
Mapping done
Merging results
Correcting cell barcodes
Looking for a whitelist
Collapsing cell barcodes
Correcting umis
Traceback (most recent call last):
  File "/home/antariksh.tyagi/miniconda3/bin/CITE-seq-Count", line 8, in <module>
    sys.exit(main())
  File "/home/antariksh.tyagi/miniconda3/lib/python3.8/site-packages/cite_seq_count/__main__.py", line 435, in main
    ) = processing.correct_umis(
  File "/home/antariksh.tyagi/miniconda3/lib/python3.8/site-packages/cite_seq_count/processing.py", line 229, in correct_umis
    for TAG in final_results[cell_barcode]:
RuntimeError: dictionary keys changed during iteration

real	516m47.259s
user	1047m20.505s
sys	28m26.428s


I tried  both python 3.6 and python 3.8 
 Thanks 

[slurm-9277.txt](https://github.com/Hoohm/CITE-seq-Count/files/5312901/slurm-9277.txt)
